source/Kepler-System-Tests/SystemHotSwapperTest.class.st

```diff
@@ -4,73 +4,163 @@ SystemHotSwapper test case
 Class {
 	#name : #SystemHotSwapperTest,
 	#superclass : #TestCase,
+	#instVars : [
+		'composite'
+	],
 	#category : #'Kepler-System-Tests'
 }
 
-{ #category : #tests }
-SystemHotSwapperTest >> testCantSwapWhenTheInterfaceIsNotImplemented [
+{ #category : #private }
+SystemHotSwapperTest >> assert: aSubsystem implements: anInterfaceKey [
 
-	| composite cms newCMS |
+	self assert: ( self is: anInterfaceKey implementedBy: aSubsystem )
+]
 
-	cms := SampleCustomerSystem new.
+{ #category : #private }
+SystemHotSwapperTest >> deny: aSubsystem implements: anInterfaceKey [
+
+	self deny: ( self is: anInterfaceKey implementedBy: aSubsystem )
+]
+
+{ #category : #private }
+SystemHotSwapperTest >> is: anInterfaceKey implementedBy: aSubsystem [
+
+	^ aSubsystem implements: ( composite interfaceAt: anInterfaceKey )
+]
+
+{ #category : #running }
+SystemHotSwapperTest >> setUp [
+
+	super setUp.
 	composite := CompositeSystem new
-		register: cms;
-		yourself.
-	composite startUp.
+]
 
-	newCMS := FixedCustomerSystem new.
+{ #category : #running }
+SystemHotSwapperTest >> tearDown [
+
+	composite shutDown.
+	super tearDown
+]
+
+{ #category : #tests }
+SystemHotSwapperTest >> testCantSwapWhenTheInterfaceIsNotImplemented [
+
+	composite
+		register: SampleCustomerSystem new;
+		startUp.
 
 	self
-		should: [ (SystemHotSwapper swapSystemImplementing: #CustomerManagementSystem with: newCMS) applyTo: composite ]
+		should: [ SystemHotSwapper swapSystemImplementing: #CustomerManagementSystem with: FixedCustomerSystem new ]
 		raise: SystemControlError
 		withMessageText: 'CMS is not implementing Customer Management'
 ]
```

Comment:
If the system does not implement the interface, it would be better to say so in the name.
Also, the error message should be "CMS does not conform to Customer Management interface" (or implementation). Implementing here might suggest that there is a single thing it should do, instead of a protocol. Another option could be "CMS does not implement all the protocol expected for Customer Management"